### PR TITLE
French as new doc language

### LINKED ISSUE #453: Introduce french translation
Hello,

While the Cookbook is entirely available in french, Authentication plugin is only available in english and japanese.
Shall I introduce the french translation ?
Please indicate whether it would be appropriate (might someone else be already working on it ?), and whether I should begin upon any basis made by some maintainer (like folders creation, etc), or simply copy/translate rst files in "fr" directory and push them up.
I use git frequently, but I'm still quite a beginner with Github so I may have missed the good way for this request. Apologize, in this case.

### PR DESCRIPTION
**Pull request by zachee54**
Resolves #453

I'm introducing a full french translation into the documentation. Please check especially that the docs/config/ change is allright.
Beside the translation itself, I made also two very tiny fixes in english version on the fly.

### CODE REVIEW

File: docs/fr/identifiers.rst [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+Confronte les identifiants fournis avec un serveur LDAP. Cet identificateur
+nécessite l'extention PHP LDAP.
+
+-  **fields**: Les champs à regarder. Par défaut
+   ``['username' => 'username', 'password' => 'password']``.
+-  **host**: Le nom complet de domaine (FQDN) de votre serveur LDAP.
+-  **port**: Le port de votre serveur LDAP. Par défaut ``389``.
+-  **bindDN**: Le nom distinctif (<em>Distinguished Name</em>) de l'utilisateur
+   à identifier. Doit être <em>callable</em>. Les <em>binds</em> anonymes de
```
Comment (markstory) [inline docs/fr/identifiers.rst:87]:
```suggestion
   à identifier. Doit être *callable*. Les *binds* anonymes de
```

File: docs/fr/identity-object.rst [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+           'id' => 'uid',
+           'username' => 'prenom'
+       ]
+   ]);
+
+Créer votre propre Objet Identité
+---------------------------------
+
+Par défaut le plugin Authentication va envelopper les données utilisateur
+renvoyées dans un ``IdentityDecorator`` qui mandate (<em>proxy</em>) l'accès aux
```
Comment (markstory) [inline docs/fr/identity-object.rst:60]:
```suggestion
renvoyées dans un ``IdentityDecorator`` qui mandate (*proxy*) l'accès aux
```

File: docs/fr/migration-from-the-authcomponent.rst [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+   Identité.
+-  Les informations sur l'utilisateur ne sont plus disponibles en passant par
+   l'ancien AuthComponent, mais sont accessibles <em>via</em> un attribut de la
+   requête et encapsulées dans un objet Identité:
+   ``$request->getAttribute('authentication')->getIdentity();``.
+   En complément, vous pouvez exploiter les méthodes ``getIdentity()`` ou
+   ``getIdentityData()`` de ``AuthenticationComponent``.
+-  La logique du processus d'authentification a été scindée en authentificateurs
+   et identificateurs. Un authentificateur va extraire les identifiants de
+   l'utilisateur (<em>credentials</em>) dans la requête, tandis que les
```
Comment (markstory) [inline docs/fr/migration-from-the-authcomponent.rst:29]:
```suggestion
   l'utilisateur (*credentials*) dans la requête, tandis que les
```

File: docs/fr/url-checkers.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Utilisez ce vérificateur si vous voulez utiliser la notation en tableaux du
+système de routage de CakePHP. Le vérificateur marche aussi avec les routes
+nommées (<em>named routes</em>).
+
+-  **checkFullUrl**: Pour comparer l'URL entière, y compris le protocole, l'hôte
+   et le port, ou pas. La valeur par défaut est ``false``
+
+Implémenter votre propre Vérificateur
+-------------------------------------q
```
Comment (markstory) [inline docs/fr/url-checkers.rst:38]:
```suggestion
-------------------------------------
```

File: docs/fr/url-checkers.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+   l'argument ``$loginUrls`` du vérificateur.
+
+CakeRouterUrlChecker
+--------------------
+
+Options:
+
+Utilisez ce vérificateur si vous voulez utiliser la notation en tableaux du
+système de routage de CakePHP. Le vérificateur marche aussi avec les routes
+nommées (<em>named routes</em>).
```
Comment (markstory) [inline docs/fr/url-checkers.rst:32]:
```suggestion
nommées (*named routes*).
```

File: docs/fr/password-hashers.rst [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+Legacy
+======
+
+C'est un hacheur de mots de passe pour les applications migrées de CakePHP2.
+
+Fallback
+========
+
+Le hacheur de mots de passe de repli (<em>fallback</em>) vous permet de
```
Comment (markstory) [inline docs/fr/password-hashers.rst:30]:
```suggestion
Le hacheur de mots de passe de repli (*fallback*) vous permet de
```

**Review by markstory — COMMENTED** [review]
This is looking really great, thank you! There are a few small formatting corrections that need to be applied before this can be deployed.

**Comment by zachee54** [comment]
I refered to [translator advises](https://github.com/cakephp/docs/blob/4.x/en/contributing/documentation.rst) in the CookBook : "_If you need to write an English term, wrap it in_ `<em>` _tags_". I applied it also to the latin word "via", as usual in french.
`<em>` can be replaced thankfully by asterisks, but you should consider to update the CookBook then. Which is the right convention ?

**Comment by markstory** [comment]
>  "If you need to write an English term, wrap it in <em> tags". I applied it also to the latin word "via", as usual in french.

Looks like we're having some formatting problems with that document. Using `*` will turn into em tags when rendered to HTML. I can get the reference guide updated.

**Pull request merged**